src/main/java/com/three_tech_solutions/slot_app/components/payment_processors/implementations/BeginningOfMonthPaymentProcessor.java

```diff
@@ -29,21 +29,21 @@ public LocalDateTime getExpirationDate(Student student) {
 
     @Override
     public Payment createInitialStudentPayment(Student student, int newPaymentNumber, Byte extraClasses) {
-        LocalDate today = LocalDate.now();
+        return getTodayDay() <= 10 ?
+                createStudentPayment(student, newPaymentNumber) :
+                createPaymentWithExtraClasses(student, newPaymentNumber, extraClasses);
+    }
 
-        if (today.getDayOfMonth() <= 10) {
-            return createStudentPayment(student, newPaymentNumber);
-        }
-        double amount = calculateExtraClassesAmount(student, extraClasses);
-        return new Payment(amount, PaymentStatus.EN_TERMINO, getExpirationDate(student), newPaymentNumber, student);
+    private static int getTodayDay() {
+        return LocalDate.now().getDayOfMonth();
     }
 
-    private double calculateExtraClassesAmount(Student student, Byte extraClasses) {
-        if (extraClasses == null || extraClasses <= 0) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
-                    "Debe especificar la cantidad de clases extra si la inscripción es posterior al día 10");
-        }
+    private Payment createPaymentWithExtraClasses(Student student, int newPaymentNumber, Byte extraClasses) {
+        return createPayment(student, newPaymentNumber, calculateExtraClassesAmount(student, extraClasses));
```

Comment:
Y acá podemos pasarle el parámetro como la fecha actual

```suggestion
        return createPayment(
                LocalDateTime.now(),
                student,
                newPaymentNumber,
                calculateExtraClassesAmount(student, extraClasses)
        );
```